mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -352,34 +353,66 @@ private boolean userHasLookupRightsOn(Mailbox mailbox, MailboxSession session) t
         return Optional.empty();
     }
 
-    private List<MailboxId> createMailboxesForPath(MailboxSession mailboxSession, MailboxPath sanitizedMailboxPath) throws MailboxException {
+    private List<MailboxId> createMailboxesForPath(MailboxSession mailboxSession, MailboxPath sanitizedMailboxPath) {
         // Create parents first
         // If any creation fails then the mailbox will not be created
         // TODO: transaction
+        List<MailboxPath> intermediatePaths = sanitizedMailboxPath.getHierarchyLevels(getDelimiter());
+        boolean isRootPath = intermediatePaths.size() == 1;
+        Function<MailboxPath, Stream<MailboxId>> duplicatedINBOX = duplicatedINBOXCreationBehavior(isRootPath);
+
+        return intermediatePaths
+            .stream()
+            .flatMap(Throwing.<MailboxPath, Stream<MailboxId>>function(mailbox -> manageMailboxCreation(mailboxSession, duplicatedINBOX, mailbox)).sneakyThrow())
+            .collect(Guavate.toImmutableList());
+    }
+
+    private Function<MailboxPath, Stream<MailboxId>> duplicatedINBOXCreationBehavior(boolean isRootPath) {
```

Comment:
this is really a sneaky way to not have to define a function that take a boolean as a parameter...